jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -109,15 +108,25 @@ ecma_builtin_object_object_get_prototype_of (ecma_value_t this_arg, /**< 'this'
 {
   JERRY_UNUSED (this_arg);
   ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+  bool was_object = ecma_is_value_object (arg);
 
   /* 1. */
-  if (!ecma_is_value_object (arg))
+  if (!was_object)
   {
+#ifndef CONFIG_DISABLE_ES2015_BUILTIN
+    ret_value = ecma_op_to_object (arg);
```

Comment:
`arg = ecma_op_to_object (arg);` makes the `arg = ret_value;` below unnecessary.